assets/src/edit-story/app/story/actions/useAutoSave.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import objectPick from '../../../utils/objectPick';
+import { useAPI } from '../../api';
+import { useConfig } from '../../config';
+import OutputStory from '../../../output/story';
+import { useSnackbar } from '../../snackbar';
+
+/**
+ * Creates AMP HTML markup for saving to DB for rendering in the FE.
+ *
+ * @param {import('../../../types').Story} story Story object.
+ * @param {Array<Object>} pages List of pages.
+ * @param {Object} metadata Metadata.
+ * @return {Element} Story markup.
+ */
+const getStoryMarkup = (story, pages, metadata) => {
```

Comment:
Let's extract this function to a utils file to prevent code duplication.

Reply:
Done in f4f9254.